org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/environments/ExecutionEnvironmentDescription.java

```diff
@@ -208,8 +217,11 @@ public ExecutionEnvironmentDescription(File eeFile) throws CoreException {
 	 * have already been performed when resolving property values.
 	 *
 	 * @return properties as a map of {@link String} keys and values
+	 * @deprecated use {@link #getProperty(String)} instead to obtain the value of a key.
 	 */
+	@Deprecated(forRemoval = true)
```

Comment:
why should this method be deprecated/removed? whats wrong with it?

Reply:
Since it returns the bare map, it allows to break the property encapsulation and data integrity as described before in https://github.com/eclipse-jdt/eclipse.jdt.debug/pull/282#discussion_r1327963578.

But it is just a suggestion, if you think it should stay, I'm fine with that as well.

Reply:
The way of deprecating in jdt is not easy. I suggest to avoid it.

Reply:
OK.
Reverted the deprecation but kept the replacement of the few usages so that this method is now unused in the Eclipse Platform.